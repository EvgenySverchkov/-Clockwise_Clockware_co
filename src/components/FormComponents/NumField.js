import React from "react";
import PropTypes from "prop-types";

const NumField = ({ max, min, id, value, chngHandler }) => (
  <div className="col-sm-3">
    <input
      id={id}
      className="form-control"
      type="number"
      min={min}
      max={max}
      onChange={chngHandler || null}
      value={value}
    />
  </div>
);

NumField.propTypes = {
  max: PropTypes.number.isRequired,
  min: PropTypes.number.isRequired,
  id: PropTypes.string.isRequired,
};

export default NumField;
