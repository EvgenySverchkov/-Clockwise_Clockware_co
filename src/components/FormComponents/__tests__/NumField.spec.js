import React from 'react';
import {create} from "react-test-renderer";

import NumField from "../NumField";
import ContextComponent from "../../../ContextComponent";

it("<NumField /> test which is used in forms", ()=>{
    const mockProps = {
        max: 5, 
        min: 1, 
        id: "1", 
        value: 5, 
        chngHandler: jest.fn(), 
        name: "Name"
    }
    const mockContextValue = {
        warningTooltipMsg: "Message",
        isOpenWarningTooltip: false,
        idForTooltip: 1
    }
    const component = create(
        <ContextComponent.Provider value={mockContextValue}>
            <NumField {...mockProps}/>
        </ContextComponent.Provider>
    )
    expect(component.toJSON()).toMatchSnapshot();
});