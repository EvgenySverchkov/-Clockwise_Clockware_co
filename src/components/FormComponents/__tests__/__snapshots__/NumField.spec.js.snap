// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumField /> test which is used in forms 1`] = `
<div
  className="col-sm-3"
>
  <input
    aria-describedby={null}
    className="form-control"
    id="1"
    max={5}
    min={1}
    name="Name"
    onBlur={[Function]}
    onChange={[MockFunction]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    title="Message"
    type="number"
    value={5}
  />
</div>
`;
